interfaces/cython/cantera/func1.pyx

```diff
@@ -87,3 +100,78 @@ cdef class Func1:
 
     def __copy__(self):
         raise NotImplementedError('Func1 object is not copyable')
+
+
+cdef class Tabulated1(Func1):
+    """
+    A `Tabulated1` object representing a tabulated function is defined by sample
+    points and corresponding function values. Inputs are specified either by
+    two iterable objects containing sample point location and function values,
+    or a single array that concatenates those inputs in two columns. Between
+    sample points, values are evaluated based on the keyword argument
+    'interpolation'; options are 'linear' (linear interpolation, default) or
+    'previous' (nearest previous value). Outside the sample interval, the value
+    at the closest end point is returned.
+
+    Examples for `Tabulated1` objects are::
+
+        >>> t1 = Tabulated1([[0, 2], [1, 1], [2, 0]])
+        >>> [t1(v) for v in [-0.5, 0, 0.5, 1.5, 2, 2.5]]
+        [2.0, 2.0, 1.5, 0.5, 0.0, 0.0]
+
+        >>> t2 = Tabulated1([[0, 2], [1, 1], [2, 0]], interpolation='previous')
+        >>> [t2(v) for v in [-0.5, 0, 0.5, 1.5, 2, 2.5]]
+        [2.0, 2.0, 2.0, 1.0, 0.0, 0.0]
+
+        >>> t3 = Tabulated1([0, 1, 2], [2, 1, 0])
+        >>> [t3(v) for v in [-0.5, 0, 0.5, 1.5, 2, 2.5]]
+        [2.0, 2.0, 1.5, 0.5, 0.0, 0.0]
+
+        >>> t4 = Tabulated1(np.array([0, 1, 2]), (2, 1, 0))
+        >>> [t4(v) for v in [-0.5, 0, 0.5, 1.5, 2, 2.5]]
+        [2.0, 2.0, 1.5, 0.5, 0.0, 0.0]
+    """
+
+    def __init__(self, *args, **kwargs):
+        if len(args) == 1:
+            # tabulated function (single argument)
+            arr = np.array(args[0])
+            if arr.ndim == 2:
+                if arr.shape[1] == 2:
+                    time = arr[:, 0]
+                    fval = arr[:, 1]
+                    method = kwargs.get('interpolation', 'linear')
+                    self._set_tables(time, fval, stringify(method))
```

Comment:
Any reason not to allow row-based input here as well? It seems as long as `ndim` is 2, it doesn't matter whether it is rows or columns.

Reply:
What about the ambiguity of the case where a 2x2 array is provided? I sort of prefer just having the two-argument form. That way, you can use your input if it's either rows or columns. Assuming numpy arrays, that would just be one of
```python
TabulatedFunction(x[0], x[1])
TabulatedFunction(x[:,0], x[:,1])
```

Reply:
That’s a valid point. I don’t have any objections to just requiring the two argument form. @bryanwweber ... any thoughts?

Reply:
Going back to Cantera/enhancements/#17, I believe there was the `vel_array = np.genfromtxt('velocity-trace.txt')` case, with the trace having two columns. I believe it would make sense to allow for a shorthand
```
TabulatedFunction(vel_array)
```
I.e. I'm tempted to just revert to the original implementation. I also think it may make sense to support something like `[(0, 1), (1, 1.5)]` where tuples are used to specify points.

Reply:
I personally prefer the two argument form that @speth suggested. A two column or row array is easy enough for the user to specify as two arguments.

Reply:
Done.